netbox/ipam/tests/test_ordering.py

```diff
@@ -0,0 +1,182 @@
+from django.test import TestCase
+
+from ipam.choices import IPAddressStatusChoices, PrefixStatusChoices
+from ipam.models import IPAddress, Prefix, VRF
+
+import netaddr
+
+
+class OrderingTestBase(TestCase):
+    vrfs = None
+
+    """
+    Setup the VRFs for the class as a whole
+    """
+    def setUp(self):
+        self.vrfs = (VRF(name="VRF A"), VRF(name="VRF B"), VRF(name="VRF C"))
+        VRF.objects.bulk_create(self.vrfs)
+
+    """
+    Perform the comparison of the queryset object and the object used to instantiate the queryset.
+    """
+    def _compare(self, queryset, objectset):
+        for i, obj in enumerate(queryset):
+            if isinstance(obj, Prefix):
+                self.assertEqual((obj.vrf, obj.prefix), (objectset[i]['vrf'], objectset[i]['prefix']))
+            elif isinstance(obj, IPAddress):
+                self.assertEqual((obj.vrf, obj.address), (objectset[i]['vrf'], objectset[i]['address']))
+
+
+class PrefixOrderingTestCase(OrderingTestBase):
+
+    """
+    This is for comparing the complex ordering test case
+    """
+    def _compare_complex(self, queryset, prefixes):
```

Comment:
It probably makes sense to move this to `OrderingTestBase` with `_compare()` even if it's only used in the one subclass for now.